presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -254,24 +245,26 @@ private void checkAllFuturesDone()
     }
 
     @Override
-    public CompletableFuture<LookupSource> readSpilledLookupSource(Session session, int partition)
+    public synchronized Set<Integer> getSpilledPartitions()
     {
-        SingleStreamSpiller lookupSourceSpiller = spilledLookupSources.get(partition);
-        Iterator<Page> spilledBuildPages = lookupSourceSpiller.getSpilledPages();
-
-        PagesIndex index = new PagesIndex(types, 10_000);
+        return spilledLookupSources.keySet();
+    }
 
-        while (spilledBuildPages.hasNext()) {
-            index.addPage(spilledBuildPages.next());
+    @Override
+    public synchronized void beginLookupSourceUnspilling(int consumers, Session session)
+    {
+        checkState(hasSpilled());
+        lookupSourceUnspilling.ifPresent(consumption -> checkState(consumption.getConsumersCount() == consumers));
```

Comment:
since you already have if statement for this condition, it would be cleaner to have simple 
```
if (lookupSourceUnspilling.isPresent) {
  ...
} 
else {
  ...
}
```
then to mix imperative and functional programming in one logical access.

Plus, please add meaningful `checkState` fail massage (FB guys were complaining that I wasn't doing that ;) )